From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Mon, 3 Aug 2015 11:46:42 -0400
Subject: [PATCH] Try to make sure configure.ac and grub-rpm-sort play nice.

Apparently the test for whether to use grub-rpm-sort and also the
renaming of it to grub2-rpm-sort on the runtime side weren't right.

Related: rhbz#1124074

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 configure.ac              | 2 +-
 util/grub-mkconfig_lib.in | 9 ++++++---
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 679f634ce4c..71d1056969e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1796,7 +1796,7 @@ AC_SUBST([LIBDEVMAPPER])
 AC_ARG_ENABLE([rpm-sort],
               [AS_HELP_STRING([--enable-rpm-sort],
                               [enable native rpm sorting of kernels in grub (default=guessed)])])
-if test x"$enable_rpm-sort" = xno ; then
+if test x"$enable_rpm_sort" = xno ; then
   rpm_sort_excuse="explicitly disabled"
 fi
 
diff --git a/util/grub-mkconfig_lib.in b/util/grub-mkconfig_lib.in
index 7fe3598435c..113a41f9409 100644
--- a/util/grub-mkconfig_lib.in
+++ b/util/grub-mkconfig_lib.in
@@ -33,6 +33,9 @@ fi
 if test "x$grub_mkrelpath" = x; then
   grub_mkrelpath="${bindir}/@grub_mkrelpath@"
 fi
+if test "x$grub_rpm_sort" = x; then
+  grub_rpm_sort="${sbindir}/@grub_rpm_sort@"
+fi
 
 if which gettext >/dev/null 2>/dev/null; then
   :
@@ -214,10 +217,10 @@ version_sort ()
    esac
 }
 
-if [ "x$RPMLIB" = x ]; then
+if [ "x$grub_rpm_sort" != x -a -x "$grub_rpm_sort" ]; then
+  kernel_sort="$grub_rpm_sort"
+else
   kernel_sort=version_sort
-else
-  kernel_sort="${sbindir}/grub-rpm-sort"
 fi
 
 version_test_numeric ()
